Prevent Python version mix and introduce NEST build option -Dwith-python=ON/OFF/2/3

gtrensch opened the pull request:
Proposed solution / workaround for issue #412.

@apeyser please can you have a look at it ?

apeyser commented:
Maybe an error if python3 == ON but python == OFF ? Or maybe we should just assume that people know what they're doing, since we may want to switch the default from python2 to python3 at some point, and such an error check would then make code complex.

So: :+1:

gtrensch commented:
@apeyser: I removed the "path/to/python" option from "-Dwith-python". This is not sufficient anyway. It is much better to set all cmake built-in variables required for Python. I have described this in INSTALL. I also removed the "-Dwith-python3" cmake argument again.  Instead of "path/to/python" a specific version can now be set with "-Dwith-python=ON/2/3/OFF". ON is the default. In that case cmake will chose a Python version (usually 2, the lowest installed). The cmake script ensures that all Python components are of exact same version. There is also a parameter check now.

jougs commented on INSTALL:
```diff
@@ -115,17 +121,42 @@ Configuring NEST for Distributed Simulation with MPI
+as an argument to `cmake`.
+
+`cmake` usually autodetects your Python installation.
+In some cases `cmake` v2.8.12 might not be able to localize the Python interpreter
```
Do we need the version number here, i.e. are we sure that the problem only happens with this exact version? I think the sentence is just as clear without the version.

apeyser replied:
Isn't it cmake < 3.4?

jougs commented on pynest/README.md:
```diff
@@ -21,6 +21,14 @@ PyNEST to `$(pyexecdir)`, which is often expanded as follows:
     $(prefix)/lib{,64}/pythonX.Y/site-packages/nest
 
+
+To force to usage of a specific Python version pass
```
"force to" → "force the"

jougs commented:
:+1: after fixing my minor comments. Many thanks for this contribution!

apeyser commented:
Discussed w/ @gtrensch :+1: 
Just confirm that the install is correct in saying cmake == 2.8.12, or is a cmake < 3.4

gtrensch commented:
Version number removed to "be on the safe side".

jougs commented:
This fixes #412.
